File: Connectivity/app/src/main/java/com/example/connectivity/MainActivity.java
package com.example.connectivity;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.Network;
import android.net.NetworkInfo;
import android.net.wifi.WifiManager;
import android.os.Bundle;
import android.util.Log;

import static android.net.wifi.WifiManager.WIFI_STATE_DISABLED;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.connect_to_wifi).setOnClickListener(v -> {
            WifiManager wManager = (WifiManager)this.getApplicationContext().getSystemService(Context.WIFI_SERVICE);
            if (wManager.getWifiState() == WIFI_STATE_DISABLED)
                wManager.setWifiEnabled(true); //true or false
        });

        findViewById(R.id.connect_to_mobile_data).setOnClickListener(v -> {
            
        });


        ConnectivityManager connMgr =
                (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
        boolean isWifiConn = false;
        boolean isMobileConn = false;
        for (Network network : connMgr.getAllNetworks()) {
            NetworkInfo networkInfo = connMgr.getNetworkInfo(network);
            if (networkInfo.getType() == ConnectivityManager.TYPE_WIFI) {
                isWifiConn |= networkInfo.isConnected();
            }
            if (networkInfo.getType() == ConnectivityManager.TYPE_MOBILE) {
                isMobileConn |= networkInfo.isConnected();
            }
        }
        Log.d(TAG, "Wifi connected: " + isWifiConn);
        Log.d(TAG, "Mobile connected: " + isMobileConn);
    }
}

File: AndroidPermission/app/src/main/java/com/iamcrypticcoder/androidpermission/MainActivity.kt
package com.iamcrypticcoder.androidpermission

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private val TAG = MainActivity::class.java.simpleName

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnNetworkPermission.setOnClickListener {
            val intent = Intent(this, NetworkPermissionActivity::class.java)
            startActivity(intent)
        }
        btnCameraPermission.setOnClickListener {
            val intent = Intent(this, CameraPermissionActivity::class.java)
            startActivity(intent)
        }
    }
}